tests/suites/3_share_integration/02_0000_share_view.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../../shell_env.sh
+
+echo "drop test share and database"
+echo "drop share if exists test_share" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "drop database if exists provider_db" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "drop database if exists provider_ref_db" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "drop share endpoint if exists ed" | $MYSQL_CLIENT_CONSUMER_CONNECT
+echo "drop catalog if exists share_catalog" | $MYSQL_CLIENT_CONSUMER_CONNECT
+
+echo "create provider share and database\table"
+echo "create database provider_db" | $MYSQL_CLIENT_PROVIDER_CONNECT
+
+echo "create provider view and reference database\table"
+echo "create database provider_ref_db;" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "create table provider_ref_db.t2 (a int);" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "insert into provider_ref_db.t2 values(3),(4);"  | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "create view provider_db.v as select * from provider_ref_db.t2;" | $MYSQL_CLIENT_PROVIDER_CONNECT
+
+echo "create provider share and grant access"
+echo "CREATE SHARE test_share;" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "GRANT USAGE ON DATABASE provider_db TO SHARE test_share;" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "GRANT REFERENCE_USAGE on database provider_ref_db to share test_share;" | $MYSQL_CLIENT_PROVIDER_CONNECT
+echo "GRANT SELECT on view provider_db.v to share test_share;" | $MYSQL_CLIENT_PROVIDER_CONNECT
```

Comment:
Does share view support ownership ?

Reply:
Please use bendsql do test `export BENDSQL_CLIENT_CONNECT="bendsql -uroot --host ${QUERY_MYSQL_HANDLER_HOST} --port ${QUERY_HTTP_HANDLER_PORT} --quote-style=never"`

And we need to add normal use to check privilege.

like https://github.com/datafuselabs/databend/blob/303f71342b8523de974734a66141162723a1a005/tests/suites/0_stateless/18_rbac/18_0007_privilege_access.sh#L7


Now we will check current_user's privilege https://github.com/datafuselabs/databend/blob/ed54911e13f9c60784b3c78037df80162734636a/src/query/service/src/interpreters/access/privilege_access.rs#L1086 about view_query.

if usr can access the view but can not access the view_query also can query view.

In share we should provider same logic.

Please to build this test case.

Reply:
ok